src/java/picard/vcf/LiftoverVcf.java

```diff
@@ -0,0 +1,232 @@
+                    "accompanying sqeuence dictionary (.dict file).")
+    public File REFERENCE_SEQUENCE = Defaults.REFERENCE_FASTA;
+
+    /** Filter name to use when a target cannot be lifted over. */
+    public static final String FILTER_CANNOT_LIFTOVER = "FailedLiftover";
+
+    /** Filter name to use when a target is lifted over, but the reference allele doens't match the new reference. */
+    public static final String FILTER_MISMATCHING_REF_ALLELE = "MismatchedRefAllele";
+
+    /** Filters to be added to the REJECT file. */
+    private static final List<VCFFilterHeaderLine> FILTERS = CollectionUtil.makeList(
+            new VCFFilterHeaderLine(FILTER_CANNOT_LIFTOVER, "Variant could not be lifted between genome builds."),
+            new VCFFilterHeaderLine(FILTER_MISMATCHING_REF_ALLELE, "Reference allele does not match reference genome sequence after liftover.")
+    );
+
+    private final Log log = Log.getInstance(LiftoverVcf.class);
+
+    // Stock main method
+    public static void main(final String[] args) {
+        new LiftoverVcf().instanceMainWithExit(args);
+    }
+
+    @Override protected int doWork() {
+        IOUtil.assertFileIsReadable(INPUT);
+        IOUtil.assertFileIsReadable(REFERENCE_SEQUENCE);
+        IOUtil.assertFileIsReadable(CHAIN);
+        IOUtil.assertFileIsWritable(OUTPUT);
+        IOUtil.assertFileIsWritable(REJECT);
+
+        ////////////////////////////////////////////////////////////////////////
+        // Setup the inputs
+        ////////////////////////////////////////////////////////////////////////
+        final LiftOver liftOver = new LiftOver(CHAIN);
+        final VCFFileReader in = new VCFFileReader(INPUT, false);
+
+        log.info("Loading up the target reference genome.");
+        final ReferenceSequenceFileWalker walker = new ReferenceSequenceFileWalker(REFERENCE_SEQUENCE);
+        final Map<String,byte[]> refSeqs = new HashMap<String,byte[]>();
+        for (final SAMSequenceRecord rec: walker.getSequenceDictionary().getSequences()) {
+            refSeqs.put(rec.getSequenceName(), walker.get(rec.getSequenceIndex()).getBases());
+        }
+        CloserUtil.close(walker);
+
+
+        ////////////////////////////////////////////////////////////////////////
+        // Setup the outputs
+        ////////////////////////////////////////////////////////////////////////
+        final VCFHeader inHeader = in.getFileHeader();
+        final VCFHeader outHeader = new VCFHeader(inHeader);
+        outHeader.setSequenceDictionary(walker.getSequenceDictionary());
+        final VariantContextWriter out = new VariantContextWriterBuilder().setOption(Options.INDEX_ON_THE_FLY)
+                .setOutputFile(OUTPUT).setReferenceDictionary(walker.getSequenceDictionary()).build();
+        out.writeHeader(outHeader);
+
+        final VariantContextWriter rejects = new VariantContextWriterBuilder().setOutputFile(REJECT).unsetOption(Options.INDEX_ON_THE_FLY).build();
```

Comment:
I'll change that.  At one point I was trying to output the rejects against the new genome, and didn't want to bother with sorting them in order to index them.  But then I changed to outputting them against the old genome so they are in the right order to be indexed.